lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -59,14 +60,26 @@ def build_index(self, server_config, dates):
             index += f"{idx}{d},"
         return index
 
+    def date_range(self, start: AnyStr, end: AnyStr) -> list:
+        """
+        Builds list of range of dates between start and end
+        It expects the date to look like YYYY-MM
+        """
+        y, m_start = int(start[:4]), int(start[5:])
+        y, m_end = int(end[:4]), int(end[5:])
+        date_range = []
+        for month in range(m_start, m_end + 1):
+            date_range.append(f"{y}-{month:02}")
+        return date_range
```

Comment:
Um, wow. How about using `datetime` or `dateutil` to actually parse and then pick out the year and month or use the datetime f-string formatting? This code is really hard to read, and without going searching for calls to figure out the input strings I have no idea whether it's correct.

Reply:
Also...this code can produce illegal date ranges...is that a "feature"??

Reply:
Added an assertion to check that the start_date is always less than the end_date